src/groundlight/client.py

```diff
@@ -95,24 +95,37 @@ def list_detectors(self, page: int = 1, page_size: int = 10) -> PaginatedDetecto
         obj = self.detectors_api.list_detectors(page=page, page_size=page_size)
         return PaginatedDetectorList.parse_obj(obj.to_dict())
 
-    def create_detector(self, name: str, query: str, config_name: str = None) -> Detector:
-        obj = self.detectors_api.create_detector(DetectorCreationInput(name=name, query=query, config_name=config_name))
+    def create_detector(
+        self, name: str, query: str, *, confidence: float = DEFAULT_CONFIDENCE_THRESHOLD, config_name: str = None
+    ) -> Detector:
+        obj = self.detectors_api.create_detector(
+            DetectorCreationInput(name=name, query=query, confidence_threshold=confidence, config_name=config_name)
+        )
         return Detector.parse_obj(obj.to_dict())
 
-    def get_or_create_detector(self, name: str, query: str, config_name: str = None) -> Detector:
-        """Tries to look up the detector by name.  If a detector with that name and query exists, return it.
+    def get_or_create_detector(
+        self, name: str, query: str, *, confidence: Optional[float] = None, config_name: str = None
+    ) -> Detector:
+        """Tries to look up the detector by name.  If a detector with that name, query, and confidence exists, return it.
         Otherwise, create a detector with the specified query and config.
         """
         existing_detector = self.get_detector_by_name(name)
         if existing_detector:
-            if existing_detector.query == query:
-                return existing_detector
-            else:
+            # TODO: We may soon allow users to update the retrieved detector's fields.
+            if existing_detector.query != query:
+                raise ValueError(
+                    f"Found existing detector with name={name} (id={existing_detector.id}) but the queries don't match. The existing query is '{existing_detector.query}'."
+                )
+            elif confidence is not None and existing_detector.confidence_threshold != confidence:
                 raise ValueError(
-                    f"Found existing detector with name={name} (id={existing_detector.id}) but the queries don't match"
+                    f"Found existing detector with name={name} (id={existing_detector.id}) but the confidence thresholds don't match. The existing confidence threshold is {existing_detector.confidence_threshold}."
                 )
+            else:
+                return existing_detector
 
-        return self.create_detector(name, query, config_name)
+        if confidence is None:
+            confidence = self.DEFAULT_CONFIDENCE_THRESHOLD
```

Comment:
Is it better to have the default in the SDK or in the cloud?

Reply:
I think it's better in the cloud - less duplication. I'll change it to default `None`, so the cloud holds the default.